### PR TITLE
#398 add PR template

### PR DESCRIPTION
**Pull request by chia7712**
required by #398

### CODE REVIEW

File: pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,8 @@
+### Checklist
+- [ ] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
+- [ ] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
```
Comment (vitojeng) [inline pull_request_template.md]: Does this mean that every PR should have least one reviewer?

Comment (chia7712) [reply pull_request_template.md]: > Does this mean that every PR should have least one reviewer?

yep

File: pull_request_template.md [hunk 2]
```diff
@@ -0,0 +1,8 @@
...
+- [ ] Add Assignees (of course, you are the assignee by default!)
+- [ ] Set Milestone (we usually file PR for master branch)
+- [ ] An existent issue should be related to this PR (required by #issue_number)
+- [ ] Provide unit tests (or please explain why this PR is not worth having new tests)
```
Comment (konekoya) [inline pull_request_template.md]: What do you think if we change this to->  _Add unit tests (and explain why if there's no need to add tests in PR)_

File: pull_request_template.md [hunk 3]
```diff
@@ -0,0 +1,8 @@
+### Checklist
+- [ ] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
+- [ ] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
+- [ ] Add Assignees (of course, you are the assignee by default!)
+- [ ] Set Milestone (we usually file PR for master branch)
+- [ ] An existent issue should be related to this PR (required by #issue_number)
```
Comment (konekoya) [inline pull_request_template.md]: I'd rephrase this to something like: Link to an issue(You can do so by using the **required by** keyword: required by #issue_number)

File: pull_request_template.md [hunk 4]
```diff
@@ -0,0 +1,8 @@
+### Checklist
+- [ ] [Review Contribution Guidelines](https://github.com/oharastream/ohara/blob/master/CONTRIBUTING.md)
+- [ ] Add Reviewers (see [Authors](https://github.com/oharastream/ohara#authors))
+- [ ] Add Assignees (of course, you are the assignee by default!)
+- [ ] Set Milestone (we usually file PR for master branch)
```
Comment (konekoya) [inline pull_request_template.md]: _Set a Milestone_ :)

File: pull_request_template.md [hunk 5]
```diff
@@ -0,0 +1,8 @@
...
+- [ ] An existent issue should be related to this PR (required by #issue_number)
+- [ ] Provide unit tests (or please explain why this PR is not worth having new tests)
+- [ ] Ensure QA passes (Sometimes you may encounter a existent flaky which obstruct you from getting green QA. Fell free to say it is not your business)
```
Comment (konekoya) [inline pull_request_template.md]: Pass QA (Sometimes you may encounter an existing flaky test which would obstruct you from getting QA passed. if the failed build is unrelated, you can add a comment in the PR thread)

Comment (chia7712) [reply pull_request_template.md]: this document is for both committer and contributor. Hence, I remove the "and merge the PR anyway"

Comment (konekoya) [reply pull_request_template.md]: Good catch :)
